src/upload.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from "react";
+import { combineReducers, applyMiddleware } from "redux";
+import { connect, ConnectedProps } from "react-redux";
+const LogRocket = require("ui/utils/logrocket").default;
+import { isDevelopment, skipTelemetry } from "ui/utils/environment";
+import { sanityCheckMiddleware } from "ui/utils/sanitize";
+const configureStore = require("devtools/client/debugger/src/actions/utils/create-store").default;
+import reducer from "ui/reducers/app";
+import { setExpectedError } from "ui/actions/session";
+import { useGetRecording, useHasExpectedError } from "ui/hooks/recordings";
+import { BlankLoadingScreen } from "ui/components/shared/BlankScreen";
+import UploadScreen from "ui/components/UploadScreen";
+
+const url = new URL(window.location.href);
+const recordingId = url.searchParams.get("id")!;
+
+function _UploadScreenWrapper({ setExpectedError }: PropsFromRedux) {
+  const expectedError = useHasExpectedError(recordingId);
+  const { recording } = useGetRecording(recordingId);
+
+  useEffect(() => {
+    if (expectedError) {
+      setExpectedError(expectedError);
+    }
+  }, [expectedError]);
+
+  useEffect(() => {
+    if (recording?.isInitialized) {
+      window.onbeforeunload = null;
+      document.location.reload();
+    }
+  });
+
+  if (expectedError) {
+    return null;
```

Comment:
Small nit - let's render <BlankLoadingScreen /> here. Otherwise, it goes `white screen → blank loading screen → white screen (null) → error`. Instead, it should be `white screen → blank loading screen → error`.